config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/EmbeddedStoragePersistServiceImpl.java

```diff
@@ -730,7 +730,8 @@ public boolean batchRemoveAggr(final String dataId, final String group, final St
     
     @Override
     public void removeConfigHistory(final Timestamp startTime, final int limitSize) {
-        String sql = "DELETE FROM his_config_info WHERE gmt_modified < ? LIMIT ?";
+        String sql = "DELETE FROM his_config_info WHERE id IN( "
+                + "SELECT id FROM his_config_info WHERE gmt_modified < ? OFFSET 0 ROWS FETCH NEXT ? ROWS ONLY)";
```

Comment:
Why not replace Limit to OFFSET 0 ROWS FETCH NEXT ? ROWS ONLY?

Reply:
The original SQL 's LIMIT will be replace to `OFFSET 0 ROWS FETCH NEXT ? ROWS ONLY`, and the delete by page cannot work in Derby. So I change the way to delete SQL.